env_store.go

```diff
@@ -127,14 +147,31 @@ func (s *EnvStore) LoadProviderSecrets(providerPaths map[string][]string) (map[s
 	return providerSecrets, nil
 }
 
+// Workaround for openBao, essentially loading secretes from Vault first.
+func (s *EnvStore) workaroundForBao(vaultPaths []string) ([]provider.Secret, error) {
+	var secrets []provider.Secret
+
+	provider, err := newProvider(vault.ProviderName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create provider %s: %w", vault.ProviderName, err)
+	}
+
+	secrets, err = provider.LoadSecrets(context.Background(), vaultPaths)
```

Comment:
we should be able to pass the context down to all methods called by the provider. lets make an issue for this @csatib02 "allow passing contexts to providers"

Reply:
#118